pom.xml

```diff
@@ -1,99 +1,81 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
 	<modelVersion>4.0.0</modelVersion>
 	<parent>
 		<groupId>org.springframework.boot</groupId>
 		<artifactId>spring-boot-starter-parent</artifactId>
-		<version>2.7.2</version>
+		<version>3.0.0-SNAPSHOT</version>
 		<relativePath/> <!-- lookup parent from repository -->
 	</parent>
 	<groupId>dev.datainmotion.airquality</groupId>
 	<artifactId>airquality</artifactId>
 	<version>2.0.0</version>
 	<name>airquality</name>
-	<description>spring-pulsar-airquality: Demo project for Spring Boot with Spring Pulsar</description>
+	<description>spring-pulsar-airquality: Demo project for Spring Boot with Spring Pulsar
+	</description>
 	<properties>
 		<java.version>17</java.version>
 		<pulsar.version>2.10.1</pulsar.version>
 	</properties>
-	<repositories>
-
-    <repository> 
-        <id>repository.spring.snapshot</id> 
-        <name>Spring Snapshot Repository</name> 
-        <url>https://repo.spring.io/snapshot</url> 
-    </repository>
-
-</repositories>
 	<dependencies>
-
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-integration</artifactId>
+			<groupId>org.springframework.pulsar</groupId>
+			<artifactId>spring-pulsar-spring-boot-starter</artifactId>
+			<version>0.1.0-SNAPSHOT</version>
 		</dependency>
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter</artifactId>
-		</dependency>
-              <dependency>
-                        <groupId>org.springframework.pulsar</groupId>
-                        <artifactId>spring-pulsar</artifactId>
-                        <version>0.1.0-SNAPSHOT</version>
-                </dependency>
-        <dependency>
-                        <groupId>org.springframework.pulsar</groupId>
-                        <artifactId>spring-pulsar-spring-boot-starter</artifactId>
-                        <version>0.1.0-SNAPSHOT</version>
-                </dependency>
-
-		<dependency>
-			<groupId>org.springframework.pulsar</groupId>
-			<artifactId>spring-pulsar-spring-boot-autoconfigure</artifactId>
-			<version>0.1.0-SNAPSHOT</version>
+			<artifactId>spring-boot-configuration-processor</artifactId>
```

Comment:
This is what generates [the metadata](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#appendix.configuration-metadata) used by tools such as IDEs to auto-complete the properties when editing the yml for instance.